src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2203,5 +2203,87 @@ private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses
             Debug.Assert(index <= maxExactClasses);
             return index;
         }
+
+        private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byte* buffer, int bufferSize)
+        {
+            FieldDesc field = HandleToObject(fieldHandle);
+            if (field.IsStatic && !field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
+            {
+                // Currently, jit is able to handle only these types of objects:
+                // * primitives
+                // * frozen strings
+                // * null
+                //
+                // To add support for mutable objects see https://github.com/dotnet/runtime/pull/76135#issuecomment-1257263126
+
+                switch (field.FieldType.Category)
+                {
+                    case TypeFlags.Boolean:
```

Comment:
CoreCLR VM also handles IntPtr/UIntPtr.

Might want to also handle `TypeFlag.Interface` (e.g. `static readonly IEnumerable<char> mystring = "Some string";`).

Feels like this would be better without the switch - just return the bytes if we have them (including arbitrary valuetypes), or reference types that are not `String`. The JIT can decide what to do with them - it's less code here that way and JIT needs to break down into the individual cases anyway (no way around that code).

Reply:
Thanks, good idea, addressed and checked that `static readonly IEnumerable<char>` works